### ISSUE TITLE
The output file rads.64.nc contails only zero values

### ISSUE THREAD

**Issue by sarriegi** [noaa-ocs-modeling/CoastalApp#181]
Hello, in all my ADCIRC-WW3 runs the output file rads.64.nc file contains only zero values. I have checked that the wave coupling works normally (the significant wave height values are normal and the water level values are different compared to the ADCIRC standalone run). So it seems that it's just an output issue. My fort.15 file looks like this:
 .
![image](https://github.com/noaa-ocs-modeling/CoastalApp/assets/77630268/cdb320a2-94de-4035-93cc-30880110438e)
 Do you have any idea of what might be going wrong?

Angelos

@pvelissariou1 @saeed-moghimi-noaa

**Comment by saeed-moghimi-noaa** [comment]
Hi @sarriegi 

Please check adcirc cap, there might be an statement to overwrite that field. In that case you may comment those lines out otherwise you may dig deeper. @pvelissariou1 may have other suggestions.

Good luck,
-Saeed

**Comment by pvelissariou1** [comment]
@sarriegi Hi Angelos. I think that ADCIRC writes the wave data to rads.64 only when swan is compiled with adcirc (e.g., adcswan). Can you share your fort.15 file?

**Comment by sarriegi** [comment]
Thank you for your responses @saeed-moghimi-noaa and @pvelissariou1 ! I am attaching the fort.15 file.

Angelos
[fort15.txt](https://github.com/noaa-ocs-modeling/CoastalApp/files/13995504/fort15.txt)

**Comment by sarriegi** [comment]
Hello @saeed-moghimi-noaa, I found these lines that concern the rads.64.nc file:
![image](https://github.com/noaa-ocs-modeling/CoastalApp/assets/77630268/81e6d7d9-2c4c-440e-bcd4-7df30bab012a)
And
![image](https://github.com/noaa-ocs-modeling/CoastalApp/assets/77630268/43d25933-deb8-471e-94fd-30203b4b95bd)
Do you know what should I do? Should I uncomment the two ALLOCATE lines in the first picture? Or set the flag in the second picture? Or neither?

Angelos

**Comment by pvelissariou1** [comment]
@sarriegi I don't think you can achieve anything by modyfing the ADCIRC code as is. I believe that data in the rads.64 file are written when CSWAN is active (I didn't go through the code thoroughly). The file is generated but fields are not written. In your case the radiation stresses are produced outside SWAN. An approach is to dump the ESMF fields fron the NUOPC cap. Saeed has implemented a "dump" option for the exported fields. Can you look at this in the NUOPC cap?

**Comment by sarriegi** [comment]
Hello @pvelissariou1 , the only file I see a mention of rads.64 (with 'grep -R rads.64') is the write_output.F file of ADCIRC. The dump option would be in adc_cap.F90?

Angelos

**Comment by pvelissariou1** [comment]
@sarriegi Yes the dump option is in adc_cap.F90 and in the subroutine State_GetFldPtr_
For your radiation stresses (sxx,syy,sxy) you call in adc_cap.F90 you have:
        call State_getFldPtr_(ST=importState,fldname='sxx',fldptr=dataPtr_sxx, &
          rc=rc,**dump=.false.**,timeStr=timeStr)
Try dump=.true.
for syy and sxy stress blocks switch  State_getFldPtr to  State_getFldPtr_ to get the "dump" option.
Play with these to see what happens

**Comment by sarriegi** [comment]
Thank you so much!

Angelos

**Comment by sarriegi** [comment]
Hello @pvelissariou1 and @saeed-moghimi-noaa, I wanted to let you know that I was able to get the rads.64.nc to show non-zero values, by modifying the ADCIRC/src/timestep.F file. More specifically, I modified the IF stametemtn in line 698:
![image](https://github.com/noaa-ocs-modeling/CoastalApp/assets/77630268/9f833524-a970-428b-92b1-614078e87a12)
Adding the .OR. ABS(NRS).EQ.5 to the IF statement.

Thank you so much for your help!

Angelos

**Comment by pvelissariou1** [comment]
@sarriegi Great, thank you for letting us know.
